sources/packages/backend/apps/queue-consumers/src/services/metrics/metrics.service.ts

```diff
@@ -0,0 +1,156 @@
+import { getQueueToken } from "@nestjs/bull";
+import { Injectable, LoggerService } from "@nestjs/common";
+import { ModuleRef } from "@nestjs/core";
+import { QueueService } from "@sims/services/queue";
+import { processInParallel } from "@sims/utilities";
+import { InjectLogger } from "@sims/utilities/logger";
+import {
+  DEFAULT_METRICS_APP_LABEL,
+  QueuesMetricsEvents,
+  MonitoredQueue,
+} from "./metrics.models";
+import { Queue } from "bull";
+import { register, collectDefaultMetrics, Gauge, Counter } from "prom-client";
+
+@Injectable()
+export class MetricsService {
+  /**
+   * List of queues to be monitored. This list is loaded once
+   * and is expected to change only if the application restarts.
+   */
+  private monitoredQueueProviders: MonitoredQueue[];
+  /**
+   * Current total number of job counts for 'active', 'completed', 'failed', 'delayed', 'waiting'.
+   * The information is acquired from Redis.
+   */
+  private readonly jobCountsGauge: Gauge;
+  /**
+   * Queue local events counter.
+   * The information is kept in memory till collected.
+   */
+  private readonly jobsEventsCounter: Counter;
+
+  /**
+   * Initializes the service and sets up the job counts gauge and jobs events counter metrics.
+   * @param moduleRef reference to the module for dependency injection.
+   * @param queueService service to interact with queue configurations.
+   */
+  constructor(
+    private readonly moduleRef: ModuleRef,
+    private readonly queueService: QueueService,
+  ) {
+    this.jobCountsGauge = new Gauge({
+      name: "queue_job_counts_current_total",
+      help: "Current total number of job counts for 'active', 'completed', 'failed', 'delayed', 'waiting'.",
+      labelNames: ["queueName", "queueEvent", "queueType"] as const,
+      collect: () => this.refreshJobCountsMetrics(),
+    });
+    this.jobsEventsCounter = new Counter({
+      name: "queue_event_total_count",
+      help: "Total number of the events for a queue If it is a global event, it will be emitted for every queue-consumer.",
+      labelNames: ["queueName", "queueEvent", "queueType"] as const,
+    });
+  }
+
+  /**
+   * Set global metrics configurations.
+   */
+  setGlobalMetricsConfigurations(): void {
+    register.setDefaultLabels({ app: DEFAULT_METRICS_APP_LABEL });
+    collectDefaultMetrics({ labels: { app: DEFAULT_METRICS_APP_LABEL } });
```

Comment:
for collect default metrics, is it not necesscary to set the register. 

```ts
collectDefaultMetrics({
      labels: { app: DEFAULT_METRICS_APP_LABEL, register },
    });
```

Reply:
Yes, there is no need to set the register unless using a custom one.

![image](https://github.com/user-attachments/assets/4351f288-9f0f-423d-a6e7-8c0e965fe323)
https://github.com/siimon/prom-client?tab=readme-ov-file#default-metrics